containers-tests/tests/intmap-strictness.hs

```diff
@@ -87,6 +87,24 @@ pInsertLookupWithKeyValueStrict f k v m
                      not (isBottom $ M.insertLookupWithKey (const3 1) k bottom m)
     | otherwise    = isBottom $ M.insertLookupWithKey (apply3 f) k bottom m
 
+pFilterWithKey :: Fun (Int, Int) Bool -> IMap -> Property
+pFilterWithKey fun m =
+  valid m' .&&. toList m' === Prelude.filter (apply fun) (toList m)
+  where
+    m' = filterWithKey (apply2 fun) m
+
+-- pFilterKeys :: Fun (Int, Int) Bool -> IMap -> Property
+-- pFilterKeys fun m =
+--   valid m' .&&. toList m' === Prelude.filter (apply fun) (toList m)
+--   where
+--     m' = filterKeys (apply2 fun) m
+
+-- pFilter :: Fun (Int, Int) Bool -> IMap -> Property
+-- pFilter fun m =
+--   valid m' .&&. toList m' === Prelude.filter (apply fun) (toList m)
+--   where
+--     m' = filter (apply2 fun) m
```

Comment:
Did you mean to add these? Currently they are commented out.

Reply:
Yes they are supposed to be added because of @treeowl his comment

> We seem to be missing property tests for filterWithKey and filter as well as the new function

But these function pFilterWithKey, pFilterKeys and pFilter were difficult for me to implement. The proposed implementation up above does not work, possible as simple as missing imports. I'll have another look if i can make something decent here otherwise i need help with these testing functions.